builder/frameworks/arduino.py

```
@@ -28,250 +28,635 @@
     call_compile_libs()
 
-if "arduino" in env.subst("$PIOFRAMEWORK") and "espidf" not in env.subst("$PIOFRAMEWORK") and env.subst("$ARDUINO_LIB_COMPILE_FLAG") in ("Inactive", "True"):
+# Main logic for Arduino Framework
+pioframework = env.subst("$PIOFRAMEWORK")
+arduino_lib_compile_flag = env.subst("$ARDUINO_LIB_COMPILE_FLAG")
+
+if ("arduino" in pioframework and "espidf" not in pioframework and 
+    arduino_lib_compile_flag in ("Inactive", "True")):
+    
+    if flag_custom_component_remove:
+        from component_manager import ComponentManager
```

Ruff (0.11.9)

`component_manager.ComponentManager` imported but unused

Remove unused import: `component_manager.ComponentManager`

(F401)